src/Features/Core/Portable/NavigateTo/NavigateToSearcher.cs

```diff
@@ -58,21 +71,73 @@ internal partial class NavigateToSearcher
                 var activeId = documentService.TryGetActiveDocument();
                 _currentDocument = activeId != null ? _solution.GetDocument(activeId) : null;
             }
+
+            var docTrackingService = _solution.Workspace.Services.GetService<IDocumentTrackingService>() ?? NoOpDocumentTrackingService.Instance;
+
+            // If the workspace is tracking documents, use that to prioritize our search
+            // order.  That way we provide results for the documents the user is working
+            // on faster than the rest of the solution.
+            _activeDocument = docTrackingService.GetActiveDocument(_solution);
+            _visibleDocuments = docTrackingService.GetVisibleDocuments(_solution)
+                                                  .WhereAsArray(d => d != _activeDocument);
```

Comment:
this just moved up.  it is needed in multiple codepaths, and can be computed synchronously.  so it can just be done in the searcher itself.